src/textfile_gui.cpp

```diff
@@ -344,48 +339,48 @@ static void Xunzip(byte **bufp, size_t *sizep)
 	FILE *handle = FioFOpenFile(textfile, "rb", dir, &filesize);
 	if (handle == nullptr) return;
 
-	this->text = ReallocT(this->text, filesize);
-	size_t read = fread(this->text, 1, filesize, handle);
+	char *buf = MallocT<char>(filesize);
```

Comment:
Replace this stuff with new/delete while you're at it?

Reply:
I tried to minimise this PR, as my first attempts were a nightmare .. the decompress functions also do magic, it really wants to realloc, and brrrr .. so I rather keep that for another day. Ideally while fixing fread too.